File: 713-subarray-product-less-than-k/subarray-product-less-than-k.py
class Solution(object):
    def numSubarrayProductLessThanK(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        n= len(nums)
        cnt=0
        i=0
        prod = 1

        for j in range(n):
            prod*=nums[j]

            while prod>=k:
                prod = prod/nums[i]
                i+=1
            cnt += j-i+1
        return cnt

File: 713-subarray-product-less-than-k/subarray-product-less-than-k.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int n = nums.size();

        int cnt = 0;
        int prod = 1;
        int i = 0;

        for(int j=0;j<n;j++){
            prod*=nums[j];

            while(prod>=k){
                prod = prod/nums[i];
                i++;
            }
            cnt += j-i+1;
        }
        return cnt;
    }
};